article/app/controllers/ArticleController.scala

```diff
@@ -117,16 +121,27 @@ class ArticleController extends Controller with RendersItemResponse with Logging
 
   def renderLiveBlog(path: String, page: Option[String] = None) =
     Action.async { implicit request =>
-      mapModel(path, blocks = true, page) {// temporarily only ask for blocks too for things we know are new live blogs until until the migration is done and we can always use blocks
-        render(path, _)
+      val range = page.map(ParseBlockId.fromPageParam) match {
+        case Some(Some(id)) => Left(PageWithBlock(id)) // we know the id of a block
+        case Some(None) => Right(NotFound) // page param there but couldn't extract a block id
+        case None => Left(Canonical) // no page param
+      }
+      range.left.map { range =>
+        mapModel(path, range = Some(range)) {// temporarily only ask for blocks too for things we know are new live blogs until until the migration is done and we can always use blocks
+          render(path, _)
+        }
+      } match {
+        case Left(f) => f
+        case Right(status) => Future.successful(Cached(10)(WithoutRevalidationResult(status)))
```

Comment:
This look confusing to me. We map page, match it, do something with the left side and match it again. Why not having only one match. Something like this:
```
      def renderWithRange(range: BlockRange) = mapModel(path, Some(range)) {
        render(path, _)
      }
      page.map(ParseBlockId.fromPageParam) match {
        case None => // no page param
          renderWithRange(Canonical)
        case Some(Some(id)) => // we know the id of a block
          renderWithRange(PageWithBlock(id))
        case Some(None) => // page param there but couldn't extract a block id
          Future.successful(Cached(10)(WithoutRevalidationResult(NotFound)))
      }
    }
```